cql3/expr/expression.cc

```diff
@@ -1779,13 +1779,35 @@ cql3::raw_value evaluate(const binary_operator& binop, const evaluation_inputs&
     return cql3::raw_value::make_value(boolean_type->decompose(binop_result));
 }
 
+cql3::raw_value evaluate(const conjunction& conj, const evaluation_inputs& inputs) {
+    for (const expression& child : conj.children) {
+        cql3::raw_value child_val = evaluate(child, inputs);
+        if (child_val.is_null()) {
+            // We could treat NULL as false, but this sounds wrong.
+            // The user can always just write `something IS NOT NULL`.
+            throw exceptions::invalid_request_exception("NULL value found inside AND conjunction");
+        }
+        if (child_val.is_unset_value()) {
+            throw exceptions::invalid_request_exception("UNSET_VALUE found inside AND conjunction");
+        }
+        if (child_val.is_empty_value()) {
+            throw exceptions::invalid_request_exception("empty value found inside AND conjunction");
+        }
+        bool bool_child_val = child_val.view().deserialize<bool>(*boolean_type);
+        if (!bool_child_val) {
+            return cql3::raw_value::make_value(boolean_type->decompose(false));
+        }
```

Comment:
I'm not sure if we're allowed to short-circuit, due to the NULL treatment. `FALSE AND NULL` should probably evaluate to NULL.

PstgresQL does evaluate these as false, so I think we can too:

```console
$ psql
psql (14.3)
Type "help" for help.

avi=# SELECT FALSE AND NULL;
 ?column? 
----------
 f
(1 row)

avi=# SELECT NULL AND FALSE;
 ?column? 
----------
 f
(1 row)

```

Reply:
Strangely, `SELECT NULL AND NULL` yields NULL.